src/App.tsx

```diff
@@ -1,61 +1,141 @@
+/* eslint-disable no-console */
+import React, { useState } from 'react';
 import './App.scss';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { TodoList } from './components/TodoList/TodoList';
+import { User } from './types/User';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+const getUserById = (id: number) => {
+  return usersFromServer.find(user => user.id === id) || null;
+};
+
+const todos = todosFromServer.map(todo => ({
+  ...todo,
+  user: usersFromServer.find(user => user.id === todo.userId) || null,
+}));
+
+export const App: React.FC = () => {
+  const [selectedUser, setSelectedUser] = useState<User | null>(null);
+  const [todoTitle, setTitle] = useState('');
+  const [visibleTodos, setTodos] = useState(todos);
+  const [noUsers, setNoUsers] = useState(false);
+  const [noTodos, setNoTodos] = useState(false);
+
+  const handleChangeTitle = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const { value } = event.target;
+
+    setNoTodos(false);
+    setTitle(value);
+  };
+
+  const handleChangeUser = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    const { value } = event.target;
+
+    setNoUsers(false);
+    setSelectedUser(getUserById(Number(value)));
+  };
+
+  const isCompleted = () => {
+    return false;
+  };
+
+  const handleAddTodo = (event: { preventDefault: () => void; }) => {
+    event.preventDefault();
+
+    if (!todoTitle) {
+      setNoTodos(true);
+    }
+
+    if (!selectedUser) {
+      setNoUsers(true);
+    }
+
+    if (!todoTitle || !selectedUser) {
+      return;
+    }
+
+    const titleToObj = () => {
+      return {
+        id: Math.random(),
+        title: todoTitle,
+        completed: isCompleted(),
+        userId: selectedUser.id || -1,
+        user: selectedUser,
+      };
+    };
```

Comment:
![image](https://user-images.githubusercontent.com/91411692/235629483-4b55d708-da09-482b-b0e2-1a6d2304d0b4.png)

```suggestion
    const titleToObj = () => {
        id: ...,
        title: todoTitle,
        completed: isCompleted(),
        userId: selectedUser.id || -1,
        user: selectedUser,
      };
```